skrub/_adaptive_squashing.py

```diff
@@ -0,0 +1,207 @@
+
+    Examples
+    --------
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> from skrub import AdaptiveSquashingTransformer
+
+    >>> tfm = AdaptiveSquashingTransformer()
+    >>> X = pd.Series([-np.inf, -100.0, 0.0, 8.0, 9.0, 10.0, 20.0, 100.0, np.inf, np.nan], name='improvement')
+    >>> tfm.fit_transform(X)
+    0   -3.000000
+    1   -2.871295
+    2   -0.789352
+    3   -0.090867
+    4    0.000000
+    5    0.090867
+    6    0.948683
+    7    2.820284
+    8    3.000000
+    9         NaN
+    Name: improvement, dtype: float32
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        max_absolute_value=3.0,
+        lower_quantile_alpha=0.25,
+        upper_quantile_alpha=0.75,
+    ):
+        self.max_absolute_value = max_absolute_value
+        self.lower_quantile_alpha = lower_quantile_alpha
+        self.upper_quantile_alpha = upper_quantile_alpha
+
+    def fit_transform(self, X, y=None):
+        """Fit the transformer and transform a column.
+
+        Parameters
+        ----------
+        X : Pandas or Polars series
+            The column to transform.
+        y : None
+            Unused. Here for compatibility with scikit-learn.
+
+        Returns
+        -------
+        X_out: Pandas or Polars series with shape (len(X),)
+            The transformed version of the input.
+        """
+        del y
+
+        if not isinstance(self.max_absolute_value, numbers.Number):
+            raise ValueError(
+                "max_absolute_value needs to be a number, but got"
+                f" type(max_absolute_value)={type(self.max_absolute_value)}"
+            )
+        if not isinstance(self.lower_quantile_alpha, numbers.Number):
+            raise ValueError(
+                "lower_quantile_alpha needs to be a number, but got"
+                f" type(lower_quantile_alpha)={type(self.lower_quantile_alpha)}"
+            )
+        if not isinstance(self.upper_quantile_alpha, numbers.Number):
+            raise ValueError(
+                "upper_quantile_alpha needs to be a number, but got"
+                f" type(upper_quantile_alpha)={type(self.upper_quantile_alpha)}"
+            )
+        if self.max_absolute_value <= 0 or not np.isfinite(self.max_absolute_value):
+            raise ValueError(
+                "max_absolute_value needs to be a positive finite float, but got"
+                f" {self.max_absolute_value}"
+            )
+        if not (0 <= self.lower_quantile_alpha < self.upper_quantile_alpha <= 1):
+            raise ValueError(
+                "need 0 <= lower_quantile_alpha < upper_quantile_alpha <= 1, but got"
+                f" lower_quantile_alpha={self.lower_quantile_alpha} and"
+                f" upper_quantile_alpha={self.upper_quantile_alpha}"
+            )
+
+        self.col_name_ = sbd.name(X)
+
+        if not sbd.is_numeric(X):
+            raise RejectColumn(f"Column {self.col_name_} is not numeric.")
+
+        eps = 1e-30  # todo: expose?
```

Comment:
```suggestion
        eps = 1e-30
```

no one will set it so we shouldn't expose it but just make sure we picked a good value. if needed you can use stuff like `np.finfo('float32').tiny`